src/Databases/MySQL/MaterializeMySQLSyncThread.cpp

```diff
@@ -104,19 +104,19 @@ static void checkMySQLVariables(const mysqlxx::Pool::Entry & connection)
          "OR (Variable_name = 'default_authentication_plugin' AND upper(Value) = 'MYSQL_NATIVE_PASSWORD') "
          "OR (Variable_name = 'log_bin_use_v1_row_events' AND upper(Value) = 'OFF');";
 
-    MySQLBlockInputStream variables_input(connection, check_query, variables_header, DEFAULT_BLOCK_SIZE, false, true);
+    StreamSettings mysql_input_stream_settings(settings, false, true);
+    MySQLBlockInputStream variables_input(connection, check_query, variables_header, mysql_input_stream_settings);
 
-    Block variables_block = variables_input.read();
-    if (!variables_block || variables_block.rows() != 5)
-    {
-        std::unordered_map<String, String> variables_error_message{
-            {"log_bin", "log_bin = 'ON'"},
-            {"binlog_format", "binlog_format='ROW'"},
-            {"binlog_row_image", "binlog_row_image='FULL'"},
-            {"default_authentication_plugin", "default_authentication_plugin='mysql_native_password'"},
-            {"log_bin_use_v1_row_events", "log_bin_use_v1_row_events='OFF'"}
-        };
+    std::unordered_map<String, String> variables_error_message{
+        {"log_bin", "log_bin = 'ON'"},
+        {"binlog_format", "binlog_format='ROW'"},
+        {"binlog_row_image", "binlog_row_image='FULL'"},
+        {"default_authentication_plugin", "default_authentication_plugin='mysql_native_password'"},
+        {"log_bin_use_v1_row_events", "log_bin_use_v1_row_events='OFF'"}
+    };
 
+    while (Block variables_block = variables_input.read())
+    {
```

Comment:
Here it is always maximum 5 rows in a block. Looks strange to make block reading in a while loop for just for 5 rows. It is better to leave `DEFAULT_BLOCK_SIZE` here as it was before.

Reply:
About this query, I wil modify it to `SHOW VARIABLES` for fix this [issue](https://github.com/ClickHouse/ClickHouse/issues/4006#issuecomment-817622685).